src/main/webapp/app/course/learning-paths/components/learning-path-exercise/learning-path-exercise.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, InputSignal, ViewContainerRef, effect, inject, input } from '@angular/core';
+import { CourseExerciseDetailsComponent } from 'app/overview/exercise-details/course-exercise-details.component';
+import { CourseExerciseDetailsModule } from 'app/overview/exercise-details/course-exercise-details.module';
+
+@Component({
+    selector: 'jhi-learning-path-exercise',
+    standalone: true,
+    imports: [CourseExerciseDetailsModule],
+    templateUrl: './learning-path-exercise.component.html',
+    styleUrl: './learning-path-exercise.component.scss',
+})
+export class LearningPathExerciseComponent {
+    public readonly courseId: InputSignal<number> = input.required<number>();
+    public readonly exerciseId: InputSignal<number> = input.required<number>();
+
+    private readonly viewContainerRef = inject(ViewContainerRef);
+
+    constructor() {
+        effect(() => {
+            this.viewContainerRef.clear();
+            const exerciseComponent = this.viewContainerRef.createComponent(CourseExerciseDetailsComponent);
+            exerciseComponent.instance.courseId = this.courseId();
+            exerciseComponent.instance.exerciseId = this.exerciseId();
+            exerciseComponent.instance.learningPathMode = true;
```

Comment:
It might not be immediately obvious why we instantiate the component here, rather than simply using it in the template. Could you add a comment that explains why this was necessary?